From 0863abd24b0a5ce354ec117ad7d4e9d96f660397 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Sat, 20 Oct 2018 17:21:50 -0600
Subject: [PATCH 06/19] msm8974-common: sepolicy: Label sysfs_batteryinfo nodes

* avc: denied { read } for name=type dev=sysfs ino=29686
  scontext=u:r:healthd:s0 tcontext=u:object_r:sysfs:s0 tclass=file
  permissive=1
* avc: denied { read } for name="type" dev="sysfs" ino=26349
  scontext=u:r:healthd:s0 tcontext=u:object_r:sysfs:s0 tclass=file
  permissive=1

Change-Id: Ie3098da96eeed27a9403e3c311fe011c1f359561
---
 sepolicy/common/file_contexts | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sepolicy/common/file_contexts b/sepolicy/common/file_contexts
index 85782ef..d84b92a 100644
--- a/sepolicy/common/file_contexts
+++ b/sepolicy/common/file_contexts
@@ -33,7 +33,9 @@
 /sys/bus/iio(/.*)?                                               u:object_r:sysfs_iio:s0
 /sys/class/mhl(/.*)?                                             u:object_r:sysfs_graphics:s0
 /sys/devices/.*/iio:device[0-9]+(/.*)?                           u:object_r:sysfs_iio:s0
-/sys/devices/battery.[0-9]+/power_supply/battery(/.*)?           u:object_r:sysfs_batteryinfo:s0
+/sys/devices/.*/power_supply(/.*)?                               u:object_r:sysfs_batteryinfo:s0
+/sys/devices/battery\.[0-9]+(/.*)?                               u:object_r:sysfs_batteryinfo:s0
+/sys/devices/i2c.*/.*-charger(/.*)?                              u:object_r:sysfs_batteryinfo:s0
 /sys/devices/mdp\.[0-9]+(/.*)?                                   u:object_r:sysfs_graphics:s0
 /sys/devices/platform/bcm[0-9]+_bluetooth/rfkill/rfkill0/state   u:object_r:sysfs_bluetooth_writable:s0
 /sys/devices/virtual/camera(/.*)?                                u:object_r:sysfs_camera:s0
-- 
2.17.1

